src/miral/basic_window_manager.cpp

```diff
@@ -763,23 +773,19 @@ auto miral::BasicWindowManager::display_area_for(WindowInfo const& info) const -
             // or if none overlap, find the area that is closest
             auto distance = std::max(
                 std::max(
-                    window_rect.left() - area->area.right(),
-                    area->area.left() - window_rect.right()).as_int(),
+                    rect.left() - area->area.right(),
+                    area->area.left() - rect.right()).as_int(),
                 std::max(
-                    window_rect.top() - area->area.bottom(),
-                    area->area.top() - window_rect.bottom()).as_int());
+                    rect.top() - area->area.bottom(),
+                    area->area.top() - rect.bottom()).as_int());
             if (distance < min_distance)
             {
                 best_area = area;
                 min_distance = distance;
             }
         }
     }
-
-    if (best_area)
-        return best_area.value();
-    else
-        return std::make_shared<DisplayArea>(window_rect);
```

Comment:
Perhaps `display_area_for(Rectangle)` should take a fallback rectangle argument and create a new `DisplayArea` with it instead of returning an optional. That would slightly simplify the code in both places that call it. Just a thought, feel free to disregard.

Reply:
I feel that makes for a weird signature. Maybe:
```c++
    auto area_of_display_area_for(Rectangle const& rect) const -> std::optional<Rectangle>;
...
    auto const placement_bounds = area_of_display_area_for(aux_rect).value_or(parent);
```